test/api_spec.js

```diff
@@ -115,91 +137,83 @@ describe("API Tests", function () {
         var path = '/user/bar';
-        util.add_target(proxy, path, port);
-        expect(proxy.routes[path].target).toEqual(target);
-        r.del(api_url + path, function (error, res, body) {
+
+        util.add_target(proxy, path, port, null, null, function () {
+            proxy._routes.get(path, function (route) {
+                expect(route.target).toEqual(target);
+
+                r.del(api_url + path, function (error, res, body) {
+                    expect(error).toBe(null);
+                    expect(res.statusCode).toEqual(204);
+                    expect(res.body).toEqual('');
+
+                    proxy._routes.get(path, function (deleted_route) {
+                        expect(deleted_route).toBe(undefined);
+                        done();
+                    });
+                });
+            });
+        });
+    });
+
+    it("GET /api/routes?inactive_since= with bad value returns a 400", function (done) {
+        r.get(api_url + "?inactive_since=endoftheuniverse", function (error, res, body) {
             expect(error).toBe(null);
-            expect(res.statusCode).toEqual(204);
-            expect(res.body).toEqual('');
-            expect(proxy.routes[path]).toBe(undefined);
+            expect(res.statusCode).toEqual(400);
             done();
         });
     });
 
     it("GET /api/routes?inactive_since= filters inactive entries", function (done) {
         var port = 8998;
         var path = '/yesterday';
-        util.add_target(proxy, '/yesterday', port);
-        util.add_target(proxy, '/today', port+1);
 
-        var now = new Date();
-        var yesterday = new Date(now.getTime() - (24 * 3.6e6));
-        var long_ago = new Date(1);
-        var hour_ago = new Date(now.getTime() - 3.6e6);
+        var now           = new Date();
+        var yesterday     = new Date(now.getTime() - (24 * 3.6e6));
+        var long_ago      = new Date(1);
+        var hour_ago      = new Date(now.getTime() - 3.6e6);
         var hour_from_now = new Date(now.getTime() + 3.6e6);
 
-        proxy.remove_route('/');
-
-        proxy.routes['/yesterday'].last_activity = yesterday;
-
         var tests = [
-            {
-                name: 'long ago',
-                since: long_ago,
-                expected: {},
-            },
-            {
-                name: 'an hour ago',
-                since: hour_ago,
-                expected: {'/yesterday': true},
-            },
-            {
-                name: 'the future',
-                since: hour_from_now,
-                expected: {
-                    '/yesterday': true,
-                    '/today': true
-                },
-            },
+            { name: 'long ago', since: long_ago, expected: {} },
+            { name: 'an hour ago', since: hour_ago, expected: { '/yesterday': true } },
+            { name: 'the future', since: hour_from_now, expected: { '/yesterday': true, '/today': true } }
```

Comment:
FWIW, I would prefer this kind of style change to not happen (before is better than after, for me).